### PR TITLE
obsolete/deprecation/g

### PR DESCRIPTION
**Pull request by delvedor**
As titled.

### CODE REVIEW

File: docs/modeling-guide.md [hunk 1]
```diff
@@ -516,3 +516,29 @@ export interface Request extends RequestBase {
...
+```
+
+#### `@deprecated_description`
+
+Use if an endpoint or property is deprecated and you need to add a description.
+
+```ts
+class Foo {
+  bar: string
+  /** @deprecated_description 7.0.0 */
```
Comment (sethmlarson) [inline docs/modeling-guide.md:540]: Should we document `@deprecated` and `@deprecated_description` together? Also should use prose in the `@deprecated_description` example?

File: specification/_types/query_dsl/geo.ts [hunk 2]
```diff
@@ -77,7 +77,7 @@ export class GeoPolygonPoints {
   points: GeoLocation[]
 }
 
-/** @obsolete 7.12.0 Use geo-shape instead. */
+/** @deprecated 7.12.0 Use geo-shape instead. */
```
Comment (sethmlarson) [inline specification/_types/query_dsl/geo.ts:80]: Expand this to `@deprecated_description`

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Awesome! Thanks for this. One thought, can we get a validator that the `@deprecated` tag matches a version number if given? Also a couple other todos below:

**Comment by delvedor** [comment]
@sethmlarson I've updated the `@deprecation` tag to accept both the version and an optional description, there is no need to have two distinct tags for this.

**Review by sethmlarson — APPROVED** [review]
LGTM

**Pull request merged**